Tests/eppo/DebugCallbackTests.swift

```diff
@@ -201,4 +204,130 @@ final class DebugCallbackTests: XCTestCase {
+        XCTAssertTrue(hasCorrectSize, "JSON parsing message should contain correct byte count (\(expectedJsonSize) bytes). Actual JSON size: \(actualJsonSizeFromMessage) bytes")
+        
+        // Verify persistent storage messages include byte counts
+        let storageWriteMessages = messageTexts.filter { $0.contains("Encoded configuration data") && $0.contains("bytes") }
+        XCTAssertGreaterThan(storageWriteMessages.count, 0, "Should have storage write message with byte count")
+        
+        // Verify persistent storage messages include reasonable encoded size
+        // Note: JSONEncoder produces more compact JSON (removes whitespace/formatting) so encoded size should be smaller
+        let actualEncodedSizeFromMessage = storageWriteMessages.first?.components(separatedBy: " ").first(where: { Int($0) != nil }) ?? "0"
+        let actualEncodedSize = Int(actualEncodedSizeFromMessage) ?? 0
+        let hasCompressedEncodedSize = actualEncodedSize > 0 && actualEncodedSize < expectedJsonSize
+        XCTAssertTrue(hasCompressedEncodedSize, "Storage write message should contain compressed encoded byte count (0 < \(actualEncodedSize) < \(expectedJsonSize))")
+    }
+    
+    func testDebugCallbackWithPersistentCacheDisabled() async throws {
+        // Arrange
+        var capturedMessages: [(String, Double, Double)] = []
+        let debugCallback: (String, Double, Double) -> Void = { message, elapsedMs, stepMs in
+            capturedMessages.append((message, elapsedMs, stepMs))
+        }
+        
+        let testConfigData = """
+        {
+            "flags": {},
+            "createdAt": "2023-01-01T00:00:00.000Z",
+            "environment": {"name": "test"},
+            "format": "SERVER"
+        }
+        """
+        
+        // Mock the HTTP response
+        stub(condition: isHost("fscdn.eppo.cloud")) { _ in
+            return HTTPStubsResponse(data: testConfigData.data(using: .utf8)!, statusCode: 200, headers: ["Content-Type": "application/json"])
+        }
+        
+        // Clear any existing cached configuration
+        EppoClient.resetSharedInstance()
+        ConfigurationStore.clearPersistentCache()
+        
+        // Act - Initialize with persistent cache disabled
+        let _ = try await EppoClient.initialize(
+            sdkKey: "test-sdk-key-no-cache",
+            withPersistentCache: false,
+            debugCallback: debugCallback
+        )
+        
+        // Wait for async operations to complete
+        try await Task.sleep(nanoseconds: Self.asyncWaitTimeNs)
+        
+        // Assert
+        let messageTexts = capturedMessages.map { $0.0 }
+        
+        // Verify no persistent storage operations occur
+        let hasStorageRead = messageTexts.contains { $0.contains("persistent storage read") }
+        let hasStorageWrite = messageTexts.contains { $0.contains("persistent storage write") }
```

Comment:
Should we instead be mocking out storage and making sure no writes called (vs checking log statements?)